### PR TITLE
feat: react table component

### PR DESCRIPTION
**Pull request by hamza14khan**
- Update table component storybook with improved examples.
- Update table styles.

Testing on Storybook:
Screenshot 2024-12-03 at 02 12 12

### CODE REVIEW

File: packages/react/ds/src/table/table-data.tsx [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
-export function TableData(props: React.PropsWithChildren) {
-  return <td>{props.children}</td>;
+interface TableDataProps extends React.PropsWithChildren {
+  className?: string;
+  colSpan?: number;
+}
+
+export function TableData({ children, className, colSpan }: TableDataProps) {
+  return (
+    <td className={`gi-table-td ${className || ''}`} colSpan={colSpan}>
```
Comment (ramonmulia) [inline packages/react/ds/src/table/table-data.tsx:10]: I suggest to use `cn` here.

File: packages/react/ds/src/table/table.stories.tsx [hunk 2]
```diff
@@ -8,40 +13,272 @@ import { TableHeader } from './table-header.js';
...
+          {headers?.map((heading, index) => (
+            <TableHeader key={`header-${index}`}>{heading}</TableHeader>
+          ))}
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        <TableRow>
+          <TableData
+            colSpan={headers.length}
+            className="!gi-text-center !gi-border-0 !gi-py-8 !gi-px-4"
```
Comment (ramonmulia) [inline packages/react/ds/src/table/table.stories.tsx:276]: do we need `!` in classes?

File: packages/design/tailwind/css/components.css [hunk 3]
```diff
@@ -943,3 +943,17 @@ input[type='file' i] {
...
+/* Table */
+
+.gi-table {
+  @apply gi-w-full gi-border-spacing-0 gi-border-collapse gi-text-md;
+}
+
+.gi-table-th,
+.gi-table-td {
+  @apply gi-text-left gi-align-top gi-pl-1 gi-pr-5 gi-py-2.5 gi-border-b-[gray] gi-border-b gi-border-solid gi-align-middle;
+  @apply gi-max-w-[25ch] gi-overflow-hidden gi-text-ellipsis gi-whitespace-nowrap;
```
Comment (BogdanDanilescu) [inline packages/design/tailwind/css/components.css:956]: is there a reason why we have 2 `@apply` here ?

Comment (hamza14khan) [reply packages/design/tailwind/css/components.css:953]: Just for readability, the second apply is for applying ellipsis, max characters etc. I can revert it if you guys feel uncomfortable with this 😄 

@pregno @ramonmulia

File: packages/react/ds/src/table/table-data.tsx [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
-export function TableData(props: React.PropsWithChildren) {
-  return <td>{props.children}</td>;
+interface TableDataProps extends React.PropsWithChildren {
+  className?: string;
+  colSpan?: number;
+}
+
+export function TableData({ children, className, colSpan }: TableDataProps) {
+  return (
+    <td className={cn('gi-table-td', className || '')} colSpan={colSpan}>
```
[the remark this answers is not in the document]
Comment (hamza14khan) [reply packages/react/ds/src/table/table-data.tsx:11]: Good suggestion

File: packages/design/tailwind/css/components.css (hunk 3 above)
Comment (pregno) [reply packages/design/tailwind/css/components.css:956]: yes, let's keep only one thanks

File: packages/react/ds/src/table/table.stories.tsx (hunk 2 above)
Comment (pregno) [reply packages/react/ds/src/table/table.stories.tsx:279]: Please avoid using `!important` classes.
Check #352 PR

Can you have an `EmptyState` component that goes into `TableData` slot for that case?

**Review by pregno — CHANGES_REQUESTED** [review]

**Review by pregno — APPROVED** [review]

**Review by BogdanDanilescu — APPROVED** [review]

**Review by ramonmulia — APPROVED** [review]

**Pull request merged**